engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCache.scala

```diff
@@ -30,19 +33,34 @@ class CallCache(database: CallCachingStore) {
   }
 
   private def toResultSimpletons(jobOutputs: JobOutputs): Seq[ResultSimpleton] = {
-   jobOutputs flatMap { case (lqn, jobOutput) => toResultSimpletons(jobOutput.wdlValue, lqn) } toSeq
-  }
-
-  private def toResultSimpletons(wdlValue: WdlValue, simpletonKey: String): Seq[ResultSimpleton] = wdlValue match {
-    case prim: WdlPrimitive => List(ResultSimpleton(simpletonKey, prim.valueString, wdlValue.getClass.getSimpleName))
-    case WdlArray(_, arrayValue) => arrayValue.zipWithIndex flatMap { case (arrayItem, index) => toResultSimpletons(arrayItem, s"$simpletonKey[$index]") }
-    case WdlMap(_, mapValue) => mapValue flatMap { case (key, value) => toResultSimpletons(value, s"$simpletonKey:${key.valueString}") } toSeq
-    case wdlObject: WdlObjectLike => wdlObject.value flatMap { case (key, value) => toResultSimpletons(value, s"$simpletonKey:$key") } toSeq
+    import cromwell.core.simpleton.WdlValueSimpleton._
+    jobOutputs.mapValues(_.wdlValue).simplify map {
+      case WdlValueSimpleton(simpletonKey, wdlPrimitive) => ResultSimpleton(simpletonKey, wdlPrimitive.valueString, wdlPrimitive.getClass.getSimpleName)
+    } toSeq
   }
 
   def fetchMetaInfoIdsMatchingHashes(callCacheHashes: CallCacheHashes)(implicit ec: ExecutionContext): Future[Set[MetaInfoId]] = {
     database.metaInfoIdsMatchingHashes(callCacheHashes.hashes map { hash => HashKeyAndValue(hash.hashKey.key, hash.hashValue.value) })
   }
 
   def fetchCachedResult(metaInfoId: MetaInfoId)(implicit ec: ExecutionContext) = database.fetchCachedResult(metaInfoId)(ec)
+
+  private def toSimpleton(entry: CallCachingResultSimpletonEntry): WdlValueSimpleton = {
+    val wdlType: WdlType = entry.wdlType match {
+      case "String" => WdlStringType
+      case "Int" => WdlIntegerType
+      case "Float" => WdlFloatType
+      case "Boolean" => WdlBooleanType
+      case _ => throw new RuntimeException(s"$entry: unrecognized WDL type: ${entry.wdlType}")
+    }
+    WdlValueSimpleton(entry.simpletonKey, wdlType.coerceRawValue(entry.simpletonValue).get.asInstanceOf[WdlPrimitive])
+  }
+
+  def convertToJobOutputs(cachedResult: CachedResult, taskOutputs: Seq[TaskOutput])(implicit ec: ExecutionContext): JobOutputs = {
+    //don't know how much error collection this *needs*
+    val simpletons = cachedResult.resultSimpletons map toSimpleton
```

Comment:
Can this field be called something like `simpletonEntries`?  It looks like simpletons are being converted to simpletons which is confusing